### PR TITLE
PLT-260:Fixed location config state issue

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: spectrocloud/resource_cluster_vsphere.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func resourceClusterVsphere() *schema.Resource {
 			"cluster_rbac_binding": schemas.ClusterRbacBindingSchema(),
 			"namespaces":           schemas.ClusterNamespacesSchema(),
 			"host_config":          schemas.ClusterHostConfigSchema(),
-			"location_config":      schemas.ClusterLocationSchema(),
+			"location_config":      schemas.ClusterLocationSchemaComputed(),
```
Comment (nikchern) [inline spectrocloud/resource_cluster_vsphere.go:222]: why do we need computed schema for VMware, shouldn’t we be able actually set VMware location?

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/resource_cluster_vsphere.go (hunk 1 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_vsphere.go]: get back to schemas.ClusterLocationSchema() and handled in cluster_location _config

File: spectrocloud/schemas/cluster_location_config.go [hunk 2]
```diff
@@ -6,6 +6,21 @@ func ClusterLocationSchema() *schema.Schema {
 	return &schema.Schema{
 		Type:     schema.TypeList,
 		Optional: true,
+		DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
```
Comment (nikchern) [inline spectrocloud/schemas/cluster_location_config.go:9]: I suggest extracting this function separately so we can unit-test it in the future.

Comment (nikchern) [reply spectrocloud/schemas/cluster_location_config.go:9]: And if possible implement unit tests now.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: spectrocloud/schemas/cluster_location_config.go (hunk 2 above)
Comment (SivaanandM) [reply spectrocloud/schemas/cluster_location_config.go:9]: I have created separate ticket to add unit test https://spectrocloud.atlassian.net/browse/PLT-312 
we can merge this PR

**Review by nikchern — APPROVED** [review]

**Pull request merged**